askama/src/filters/mod.rs

```diff
@@ -139,9 +139,17 @@ impl fmt::Display for FilesizeFormatFilter {
 /// To encode `/` as well, see [`urlencode_strict`](./fn.urlencode_strict.html).
 ///
 /// [`urlencode_strict`]: ./fn.urlencode_strict.html
-pub fn urlencode<T: fmt::Display>(s: T) -> Result<String> {
-    let s = s.to_string();
-    Ok(utf8_percent_encode(&s, URLENCODE_SET).to_string())
+#[inline]
+pub fn urlencode(s: impl fmt::Display) -> Result<UrlencodeFilter, Infallible> {
+    Ok(UrlencodeFilter(s.to_string()))
```

Comment:
Not using this [`percent-encoding`](https://docs.rs/percent-encoding/latest/percent_encoding/fn.percent_decode.html). The library wants to work on byte slices.

I guess we could implement the functionality ourself, but the implementor won't be me. :) If I need to encode query strings, then I'd prefer using [`serde_qs`](https://crates.io/crates/serde_qs) 9/10 times, so I can use the same datatype for serialization and deserialization.

Reply:
Ah, I did not think this through: You can simply escape substrings, and print out the result. This works because we know that we won't get truncated UTF-8 characters, or surrogate pairs.